src/main/java/org/folio/circulation/resources/renewal/RegularRenewalStrategy.java

```diff
@@ -51,11 +53,36 @@
   }
 
   private Result<RenewalContext> renew(RenewalContext context) {
+    return renew(context, DateTime.now(DateTimeZone.UTC), context.getRequestQueue())
+      .map(context::withLoan);
+  }
+
+  private Result<Loan> renew(RenewalContext context, DateTime systemDate,
+    RequestQueue requestQueue) {
+
     Loan loan = context.getLoan();
-    RequestQueue requestQueue = context.getRequestQueue();
+    try {
+      final var errors = validateIfRenewIsAllowed(loan, requestQueue);
+      final var loanPolicy = loan.getLoanPolicy();
 
-    return renew(loan, DateTime.now(DateTimeZone.UTC), requestQueue)
-      .map(context::withLoan);
+      if (loanPolicy.isNotLoanable() || loanPolicy.isNotRenewable()) {
+        return failedValidation(errors);
+      }
+
+      final Result<DateTime> proposedDueDateResult = calculateNewDueDate(loan, requestQueue,
+        systemDate);
+      addErrorsIfDueDateResultFailed(loan, errors, proposedDueDateResult);
+
+      final BlockOverrides blockOverrides = BlockOverrides.from(getObjectProperty(
+        context.getRenewalRequest(), "overrideBlocks"));
+      if (errors.isEmpty() && !blockOverrides.getPatronBlockOverride().isRequested()) {
```

Comment:
What will happen if there are no errors and patron block override is requested? Do we have a test for this scenario?